rs/nns/cmc/src/lib.rs

```diff
@@ -313,10 +315,23 @@ pub struct CyclesLedgerDepositResult {
     pub block_index: Nat,
 }
 
+// When a user sends us ICP, they indicate via memo (or icrc1_memo) what
+// operation they want to perform.
+//
+// We promise that we will NEVER use 0 as one of these values. (This would be
+// very bad, because if no memo is explicitly supplied, the Transaction might
+// implicitly have 0 in the memo field.)
+//
+// Note to developers: If you add new values, update MEANINGFUL_MEMOS.
 pub const MEMO_CREATE_CANISTER: Memo = Memo(0x41455243); // == 'CREA'
 pub const MEMO_TOP_UP_CANISTER: Memo = Memo(0x50555054); // == 'TPUP'
 pub const MEMO_MINT_CYCLES: Memo = Memo(0x544e494d); // == 'MINT'
 
+// New values might be added to this later. Do NOT assume that values won't be
+// added to this array later.
+pub const MEANINGFUL_MEMOS: [Memo; 3] =
```

Comment:
Why can't we use lazy_static to define these values in a hash map?

Something like 

```rust
lazy_static! {
    RESERVED_MEMOS: BTreeMap<String, Memo> = btreemap! {
        "CREATE_CANISTER".to_string() => Memo(0x41455243),
        // ...
    };
}
```